README.md

```diff
@@ -1,2 +1,98 @@
-# nbde_client
-Ansible role for configuring Network Bound Disk Encryption clients (e.g. clevis)
+nbde_client
+-----------
+
+Ansible role for configuring Network-Bound Disk Encryption clients (e.g. clevis).
+
+This role currently supports `clevis` as a provider and it uses it for operations like encryption
+and decryption.
+
+
+Supported Distributions
+-----------------------
+* RHEL-7+, CentOS-7+
+* Fedora
+
+
+Limitations
+-----------
+This role can currently create `tang` bindings. TPM2 is not supported as of now.
+
+
+Role Variables
+--------------
+
+These are the variables that can be passed to the role:
+
+
+| **Variable** | **Default/Choices** | **Description** |
+|----------|-------------|------|
+| `nbde_client_provider` | `clevis`| identifies the provider for the `nbde_client` role. We currently support `clevis`.|
+| `nbde_client_bindings` | | a list containing binding configurations, which include e.g. devices and slots. |
+
+
+#### nbde_client_bindings
+`nbde_client_bindings` is a list of dictionaries that support the following keys:
+| **Name** | **Default/Choices** | **Description** |
+|----------|-------------|------|
+| `device` | | specifies the path of the backing device of an encrypted device on the managed host. This device must be already configured as a LUKS device before using the role (**REQUIRED**). |
+| `passphrase` | | a valid passphrase for opening/unlocking the specified device. |
```

Comment:
If this needs to be protected in the Ansible inventory, we should include some text like this:
```
recommend vault encrypting the private key password
see https://docs.ansible.com/ansible/latest/user_guide/vault.html
```
As in the network docs - https://github.com/linux-system-roles/network/#examples-of-options

Reply:
> If this needs to be protected in the Ansible inventory, we should include some text like this:
> 
> ```
> recommend vault encrypting the passphrase
> see https://docs.ansible.com/ansible/latest/user_guide/vault.html
> ```
> 
> As in the network docs - https://github.com/linux-system-roles/network/#examples-of-options

Reply:
This is a good idea.